typescript/token/contracts/libs/FastTokenRouter.sol

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {Message} from "./Message.sol";
+import {TokenRouter} from "./TokenRouter.sol";
+
+import {TypeCasts} from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+
+/**
+ * @title Common FastTokenRouter functionality for ERC20 Tokens with remote transfer support.
+ * @author Abacus Works
+ */
+abstract contract FastTokenRouter is TokenRouter {
+    using TypeCasts for bytes32;
+    using Message for bytes;
+    /**
+     * @notice `FastTransferMetadata` is the LP data stored against `fastTransferId`.
+     */
+    struct FastTranferMetadata {
+        address filler;
+        address recipient;
+        uint256 amount;
+        uint256 fastFee;
+    }
+
+    uint256 public fastTransferId;
+    // maps `fastTransferId` to `FastTranferMetadata`.
+    mapping(bytes32 => FastTranferMetadata) filledFastTransfers;
+
+    /**
+     * @dev delegates transfer logic to `_transferTo`.
+     * @inheritdoc TokenRouter
+     */
+    function _handle(
+        uint32 _origin,
+        bytes32,
+        bytes calldata _message
+    ) internal virtual override {
+        bytes32 recipient = _message.recipient();
+        uint256 amount = _message.amount();
+        bytes calldata metadata = _message.metadata();
+        _transferTo(recipient.bytes32ToAddress(), amount, _origin, metadata);
+        emit ReceivedTransferRemote(_origin, recipient, amount);
+    }
+
+    /**
+     * @dev Transfers `_amount` of token to `_recipient`/`fastFiller` who provided LP.
+     * @dev Called by `handle` after message decoding.
+     */
+    function _transferTo(
+        address _recipient,
+        uint256 _amount,
+        uint32 _origin,
+        bytes calldata _metadata
+    ) internal virtual {
```

Comment:
I think its confusing to have the same name but not override the base class

Reply:
But this is a common design choice used right? Even the `GasRouter` implements this type of design right?
A bit reluctant to change the name because `_handle` uses `_transferTo` for the transfer logic and I wouldn't want to change that.
What do you think? @yorhodes